### ISSUE TITLE
Adding Domain Group Management page to portfolio view

### ISSUE THREAD

**Issue by zandercymatics** [cisagov/manage.get.gov#2396]
### Issue description

Domain groups are meant to act like a label on a domain. These will be set by admin users to "group" domains together to help with their own internal organization. For instance, some users may group domains that have a particular manager or who expire within the same month. Just like in github where we can filter by a label, the goal is to let users filter by any set of domain groups (labels).

To achieve this goal, we should add the domain group page to our portfolio view and link to it via to the domain group tab. This will serve as a stub for work to expand onto.

As this is a stub, only worry about implementing a read-only version of the domain group page. The create, add, and save functionalities (as well as the search bar) will be implemented later. These buttons **should** exist on the page, but they will not be linked to anything yet.

### Acceptance criteria
**The contents of the domain group page are as follows:**
- [ ] A barebones (aka readonly) table of all domain groups for the given user. 
  - For now, this view is readonly (create, save, and delete are handled under follow-on tickets). Likewise, modals will also be accounted for in these tickets.
  - The search bar will be implemented in a [follow-on ticket](https://github.com/cisagov/manage.get.gov/issues/2435). There is no need to stub it out here.
  - I'd suggest referring to the `domains_table.html` template that is used on the domains table. See the `loadDomains` function in `get-gov.js` for further implementation details.
- [ ] The "Domain groups" side nav. 
- [ ] Mobile view is accounted for

**Additionally, ensure that:**
- [ ] The domain group tab for portfolios links to the new domain group page. 
- [ ] The table header includes the "New group" button. 
  - **This button should not do anything yet.** Create functionality is reserved for a follow-on ticket.
- [ ] The table includes a "edit" and "delete button on each row.
  - **These buttons should not do anything yet.** Edit/delete functionality is reserved for a follow-on tickets.

### Additional context
- Reference **[this figma](https://www.figma.com/design/xFtGLHVrhp0lvwh0gYWnbx/.gov-Designs-CISA?node-id=4911-15641&t=FjNi8Cyb0ts3LEUS-4)** when implementing this ticket. See [Figma](https://www.figma.com/design/xFtGLHVrhp0lvwh0gYWnbx/.gov-Designs-CISA?node-id=5053-21131&t=FjNi8Cyb0ts3LEUS-4) mobile designs.
- Reminder: each domain can have multiple domain groups. **These should stack on top of each other**, rather than be comma-separated.
- Text should wrap after exceeding max length. Words should hyphenate if they break. 

Refinement note: We may consider directly splitting this ticket into three pieces: an api portion and a UI portion (then a part that combines those). However, this is implementation specific. For now, lets keep this ticket and iterate _if we need to_ rather than rely on it 

### Links to other issues

⛔ blocked by 
  - #2349
  - #2200
  - #2365 

✋ Blocks
- #2435
 - #2433
 - #2434
 - #2452 

🔄 Relates to
- #2395

**Comment by katypies** [comment]
Closing this. We don't currently have a sense of whether domain groups will be needed. We can re-open or create a new issue if this comes up in our user interviews or usability testing. I suspect that suborgs may solve many of the intended use-cases here, but it's worth waiting to determine if this is a useful addition in the future.
